[ISSUE] databricks_grants not exported for objects with only inherited permissions but referenced by other resources using depends_on

I encountered an error while running terraform validate after generating the HCL code using the Databricks provider v1.51.0. The error indicates a reference to an undeclared resource, suggesting that objects with only inherited permissions are not being rendered by the exporter, while dependencies on other objects are present. Below are the details of the configuration, expected and actual behavior, steps to reproduce, and relevant information.

Configuration
```hcl

# Example Terraform configuration that caused the issue
resource "databricks_volume" "dev_main_playground_test" {
  # Configuration details
  depends_on   = [databricks_grants.schema_dev_main_playground]
}
```
Expected Behavior
The exporter should generate HCL code with all the necessary resources, including those with inherited permissions, so that terraform validate does not produce any errors.

Actual Behavior
Received the following error during terraform validate:

```shell
Error: Reference to undeclared resource
│
│   on uc-volumes.tf line 33, in resource "databricks_volume" "dev_main_playground_test":
│   33:   depends_on   = [databricks_grants.schema_dev_main_playground]
│
│ A managed resource "databricks_grants" "schema_dev_main_playground" has not been declared in the root module
```

Steps to Reproduce
1. Run terraform-provider-databricks_v1.51.0.exe exporter -skip-interactive -services="uc-grants,uc-tables" to generate the HCL code.
2. Run terraform validate.
Terraform and provider versions
Terraform version: v1.9.5
Databricks provider version: v1.51.0
Is it a regression?
Not sure if this is a regression. This behavior was encountered using the v1.51.0 provider.

Debug Output
N/A

Important Factoids
The issue seems to be caused by objects that have only inherited permissions not being rendered by the exporter, but dependencies on these objects are still present in the generated code.
```shell
databricks grants get schema dev-main.playground
{}
```

Would you like to implement a fix?
No, I would not like to implement a fix.

all have the same owner.
What I have tried. Set a catalog owner to Service Prinicpal which I do authentication to the workspace and it helps.

[ERROR] 2024-12-05T18:07:58.7410594+01:00: listing catalog bindings for main: User is not an owner of Catalog 'main'.

--- a/NEXT_CHANGELOG.md
+++ b/NEXT_CHANGELOG.md
@@ -16,5 +16,6 @@
 ### Exporter
 
  * Correctly handle account-level identities when generating the code ([#4650](https://github.com/databricks/terraform-provider-databricks/pull/4650))
+ * Add `Ignore` implementation for `databricks_grants` to fix issue with wrongly generated dependencies ([#4661](https://github.com/databricks/terraform-provider-databricks/pull/4650))
 
 ### Internal Changes

--- a/exporter/importables.go
+++ b/exporter/importables.go
@@ -2110,6 +2110,10 @@ var resourcesMap map[string]importable = map[string]importable{
 		WorkspaceLevel: true,
 		Service:        "uc-grants",
 		Import:         importUcGrants,
+		Ignore: func(ic *importContext, r *resource) bool {
+			return (r.Data.Get("grant.#").(int) == 0)
+		},
+
 		Depends: []reference{
 			{Path: "catalog", Resource: "databricks_catalog"},
 			{Path: "schema", Resource: "databricks_schema"},

